rules/rules.js

```diff
@@ -85,61 +116,71 @@ const getGroupsForItem = function (ruleConfig) {
  *
  * rules.JSRule({
  *  name: "my_new_rule",
- *  description": "this rule swizzles the swallows",
+ *  description: "this rule swizzles the swallows",
  *  triggers: triggers.GenericCronTrigger("0 30 16 * * ? *"),
  *  execute: triggerConfig => { //do stuff }
  * });
  *
  * @memberOf rules
  * @param {Object} ruleConfig The rule config describing the rule
  * @param {String} ruleConfig.name the name of the rule
- * @param {String} ruleConfig.description a description of the rule
+ * @param {String} [ruleConfig.description] a description of the rule
  * @param {*} ruleConfig.execute callback that will be called when the rule fires
  * @param {HostTrigger|HostTrigger[]} ruleConfig.triggers triggers which will define when to fire the rule
+ * @param {String} [ruleConfig.id] the UID of the rule
+ * @param {String} [ruleConfig.ruleGroup] the name of the rule group to use
+ * @param {Boolean} [ruleConfig.overwrite=false] overwrite an existing rule with the same UID
  * @returns {HostRule} the created rule
  */
 let JSRule = function (ruleConfig) {
     let ruid = ruleConfig.id || ruleConfig.name.replace(/[^\w]/g, "-") + "-" + utils.randomUUID();
-    let ruTemplateid = ruleConfig.name.replace(/[^\w]/g, "-") + "-" + utils.randomUUID();
-    log.info("Adding rule: {}", ruleConfig.name ? ruleConfig.name : ruid);
-
-    let SimpleRule = Java.extend(Java.type('org.openhab.core.automation.module.script.rulesupport.shared.simple.SimpleRule'));
-
-    let doExecute = function (module, input) {
-        try {
-            return ruleConfig.execute(getTriggeredData(input));
-        } catch (error) {
-            log.error(`Failed to execute rule ${ruid}: ${error}: ${error.stack}`);
-            throw error;
-        }
-    };
-
-    let rule = new SimpleRule({
-        execute: doExecute,
-        getUID: () => ruid
-    });
-
-    let triggers = ruleConfig.triggers ? ruleConfig.triggers : ruleConfig.getEventTrigger();
-    if (!Array.isArray(triggers)) {
-        triggers = [triggers];
+    if (ruleConfig.overwrite === true) {
+        removeRule(ruid);
     }
-
-    rule.setTemplateUID(ruTemplateid);
-
-    if (ruleConfig.description) {
-        rule.setDescription(ruleConfig.description);
-    }
-    if (ruleConfig.name) {
-        rule.setName(ruleConfig.name);
-    }
-
-    //Register rule here
-    if (triggers && triggers.length > 0) {
-        rule.setTriggers(triggers);
-        rule = registerRule(rule);
+    if (ruleExists(ruid)) {
+        log.error('Failed to add rule: {}, a rule with same UID [{}] already exists!', ruleConfig.name ? ruleConfig.name : ruid, ruid);
```

Comment:
should we instead throw the error instead of logging it?

Reply:
Ups, yes, give me a few minutes.